### DISCUSSION TITLE
MySQL Resource Grants in Workshop

### DISCUSSION THREAD

**Discussion by nprimmer** [noop-inc/documentation#100] [Bugs] [answered]
Repo: https://github.com/nprimmer/blueprint-go-gin

I have a MySQL Resource declared as part of my template, and when creating a new version of the application in Workshop, my application gives the following errors:
```
07:52:17.892
2024/02/03 15:52:17 [35m/build/main.go:45 [0m[31m[error] [0mfailed to initialize database, got error Error 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES) [31mFATA[0m[0000] Failed to connect to database: Error 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
07:52:17.897
[31mFATA[0m[0000] 1 errors:
```

This MySQL resource is declared as follows:
```
resources:
  - name: dbserver
    type: mysql
```

The DSN components are passed to the application as environment variables as such:
```
    runtime:
      command: '/app/backend'
      variables:
        MYSQL_HOST:
          $resources: dbserver.host
        MYSQL_USER:
          $resources: dbserver.username
        MYSQL_PASSWORD:
          $resources: dbserver.password
        MYSQL_DBNAME:
          $resources: dbserver.database
      resources:
        - dbserver
```

The resulting values are:
User - root
Password - secret
Host - resource.workshop
Database - dbserver

I've validated that the application will run against a locally configured MySQL server, and the identical commit will also run in Noop Cloud without issue.

**Comment by derekwolpert** [comment, marked as answer]
Hey @nprimmer 👋

We were able to get a version of your Go Template application working in Noop Desktop by providing an explicit reference to the MySQL resource's port value. [Link to the needed code adjustments.](https://github.com/noop-inc/template-go-gin/commit/3ca38d87b9647d8a8288bcb323e8760ea914990d) Providing an explicit port value is required within Noop Desktop due to usage of non-standard port values for MySQL connections.

When you have a chance try out these changes on your end, and let us know how it goes 🤞
